rust/src/detect/transform_base64.rs

```diff
@@ -47,11 +47,16 @@ pub struct SCDetectTransformFromBase64Data {
     offset: u32,
     offset_str: *const c_char,
     mode: SCBase64Mode,
+
+    // serialized data for hashing
+    serialized: *const u8,
+    serialized_len: u32,
```

Comment:
@jasonish should SCDetectTransformFromBase64Data own this ? Makes more sense for pcrexform, and we need a common scheme...

Reply:
I think this makes sense.  Normally I'd say the serialize method should return a new object containing the data, but I believe that data then needs to travel through the system with this struct right? So makes sense it just owns it instead of making things more complicated.

Reply:
>  but I believe that data then needs to travel through the system with this struct right?

This was the design by @jlucovsky but this raises the question wether the design is right.